tests/cmaes_test.cpp

```diff
@@ -172,3 +172,26 @@ TEST_CASE("ApproxCMAESLogisticRegressionFMatTest", "[CMAESTest]")
 
   REQUIRE(success == true);
 }
+
+/**
+ * Test to check if the boundary transformation
+ * works as expected.
+ */
+TEST_CASE("BoundaryConditionTestFunction", "[CMAESTest]")
+{
+  SGDTestFunction f;
```

Comment:
Do you mind to change the object function here, to something more complex, the SGD function has only 3 parameters.

Reply:
@gaurav-singh1998 let me know if you need any help with this.

Reply:
Hi @zoq, sorry for being so late on this(life got messed up). I want to take this up again. Can you provide some help as to what objective functions I need to use and what other changes do I need to make? Thanks.

Reply:
@gaurav-singh1998 No worries I hope everything is going well. Let's test this one on the `LogisticRegression` function, here is an example - https://github.com/mlpack/ensmallen/blob/60a99f780ba24fb0403579534bc53db9e09f54c9/tests/cmaes_test.cpp#L40-L66.